parsec/core/fs/storage/chunk_storage.py

```diff
@@ -230,6 +238,14 @@ async def set_chunk(self, chunk_id: ChunkID, raw: bytes) -> None:
                 (chunk_id.bytes, len(ciphered), False, time.time(), ciphered),
             )
 
+            # Perform cleanup if necessary
+            await self.cleanup(cursor)
+
+    async def cleanup(self, cursor: Optional[Cursor] = None) -> None:
+
+        # Update database
+        async with self._reenter_cursor(cursor) as cursor:
+
             # Count the chunks
             cursor.execute("SELECT COUNT(*) FROM chunks")
```

Comment:
Not directly related to this PR, but it seems very costly to me to have this request being made every time we insert a new chunk (the query itself should be very fast, be we pay the query parsing, python-to-c boundary crossing with GIL release, SQlite lock acquire etc.). Wouldn't it be better to add the `SELECT COUNT(*) FROM chunks` as an aditional `VALUES` to the `INSERT OR REPLACE` query instead ?

Reply:
I don't think it's worth it, the previous query takes about 50 times longer so there's not much too gain:
```python
In [1]: import sqlite3, uuid, time

In [2]: path = "/tmp/tmpqh61dj4l/share/parsec/9b3bc58589#Org#4418c3e2a3144db2baa1c9ae6041ed46@31de682070ea489f9bf201cf625af8ea/e190f567-0fe3-4f
   ...: a2-b104-736752f4825f/workspace_cache-v1.sqlite"

In [3]: c = sqlite3.connect(path)

In [6]: data = b"a" * 512 * 1024

In [7]: %%time
   ...: c.execute(
   ...:   """INSERT OR REPLACE INTO
   ...:   chunks (chunk_id, size, offline, accessed_on, data)
   ...:   VALUES (?, ?, ?, ?, ?)""",
   ...:   (uuid.uuid4().bytes, len(data), False, time.time(), data),
   ...: )
   ...: 
   ...: 
CPU times: user 0 ns, sys: 3.77 ms, total: 3.77 ms
Wall time: 7.58 ms
Out[7]: <sqlite3.Cursor at 0x7eff41acac70>

In [8]: %time print(c.execute("SELECT COUNT(*) FROM chunks").fetchone())
(956,)
CPU times: user 117 µs, sys: 5 µs, total: 122 µs
Wall time: 167 µs